arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,295 @@
+    let symbolStyle = AGSSymbolStyle(styleName: "Esri2DPointSymbolsStyle", portal: .arcGISOnline(withLoginRequired: false))
+    /// The observation on the map view's `mapScale` property.
+    var mapScaleObservation: NSKeyValueObservation?
+    /// A cache for the image swatches of the symbols.
+    var cachedImages = [AGSSymbol: UIImage]()
+    /// The data source for the legend table.
+    private var symbolsDataSource: SymbolsDataSource?
+    /// A list of symbols created from the web style.
+    private var symbols = [(category: SymbolCategory, symbol: AGSSymbol)]()
+    
+    /// A feature layer with LA County Points of Interest service.
+    let featureLayer: AGSFeatureLayer = {
+        let serviceFeatureTable = AGSServiceFeatureTable(url: URL(string: "http://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/LA_County_Points_of_Interest/FeatureServer/0")!)
+        let layer = AGSFeatureLayer(featureTable: serviceFeatureTable)
+        return layer
+    }()
+    
+    // MARK: Methods
+    
+    /// Create a map with an `AGSFeatureLayer` added to its operational layers.
+    /// - Parameter featureLayer: An `AGSFeatureLayer` object.
+    /// - Returns: An `AGSMap` object.
+    func makeMap(featureLayer: AGSFeatureLayer) -> AGSMap {
+        let map = AGSMap(basemapStyle: .arcGISLightGray)
+        map.referenceScale = 1e5
+        map.operationalLayers.add(featureLayer)
+        return map
+    }
+    
+    /// Create an `AGSUniqueValueRenderer` to render feature layer with symbol styles.
+    /// - Parameter fieldNames: The attributes to match the unique values against.
+    /// - Returns: An `AGSUniqueValueRenderer` object.
+    func makeUniqueValueRenderer(fieldNames: [String]) -> AGSUniqueValueRenderer {
+        let renderer = AGSUniqueValueRenderer()
+        renderer.fieldNames = fieldNames
+        renderer.uniqueValues = symbols.flatMap { category, symbol in
+            // For each category value of a symbol, we need to create a
+            // unique value for it, so the field name matches to all categories.
+            category.symbolCategoryValues.map { symbolValue in
+                AGSUniqueValue(description: "", label: category.symbolName, symbol: symbol, values: [symbolValue])
+            }
+        }
+        return renderer
+    }
+    
+    /// Get certain categories of symbols from a symbol style.
+    /// - Parameters:
+    ///   - symbolStyle: An `AGSSymbolStyle` object.
+    ///   - categories: The types of symbols to search in the symbol style.
+    ///   - completion: A closure executed upon success.
+    private func getSymbols(symbolStyle: AGSSymbolStyle, categories: [SymbolCategory], completion: @escaping () -> Void) {
+        let getSymbolsGroup = DispatchGroup()
+        var symbols = [(SymbolCategory, AGSSymbol)]()
+        
+        categories.forEach { category in
+            getSymbolsGroup.enter()
+            symbolStyle.symbol(forKeys: [category.symbolName]) { symbol, _ in
```

Comment:
`symbolName` is computed multiple times (here, in `makeUniqueValueRenderer(fieldNames:)`, etc.), but I think that can be avoided. Calculate it once by storing it to be local variable and then use the value of that variable here and, if you take my suggestion to eliminate the image cache, when creating the legend item. Then `makeUniqueValueRenderer(fieldNames:)` can use the legend item's name directly. Does that make sense?